utils/funcexp/func_timediff.cpp

```diff
@@ -162,16 +162,35 @@ string Func_timediff::getStrVal(rowgroup::Row& row,
         case execplan::CalpontSystemCatalog::CHAR:
         case execplan::CalpontSystemCatalog::TEXT:
             text = parm[0]->data()->getStrVal(row, isNull);
-            isTime1 = std::count(text.begin(), text.end(), '-') <= 1; // Time can have at most 1 dash (signifies negative)
-            val1 = dataconvert::DataConvert::stringToDatetime(text, &isDate1);
+            
+            if (text.length() >= 14) // datetime has length at least 14
```

Comment:
This is the major difference between results for MDB and columnstore. We should be converting all non-temporal data types to times if length is less than 14. There was a special case i found while testing results. see comment on line 170

Reply:
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/957#discussion_r353731147

Reply:
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/957#discussion_r353734092

Reply:
@drrtuy Have updated magic numbers to account for all locales considered during data type conversions between string types and datetimes/times, as well as int types and datetimes/times. Please see new commit